File: androidApp/src/main/java/dev/johnoreilly/mortycomposekmm/ui/episodes/EpisodesListRowView.kt
package dev.johnoreilly.mortycomposekmm.ui.episodes

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import dev.johnoreilly.mortycomposekmm.fragment.EpisodeDetail


@Composable
fun EpisodesListRowView(episode: EpisodeDetail, episodeSelected: (episode: EpisodeDetail) -> Unit) {

    Row(modifier = Modifier.fillMaxWidth()
        .clickable(onClick = { episodeSelected(episode) })
        .padding(vertical = 8.dp, horizontal = 16.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {

        Column(modifier = Modifier.weight(1f)) {
            Text(
                episode.name,
                style = MaterialTheme.typography.h6,
                fontWeight = FontWeight.Bold,
                maxLines = 1, overflow = TextOverflow.Ellipsis,)

            CompositionLocalProvider(LocalContentAlpha provides ContentAlpha.medium) {
                Text(
                    episode.episode,
                    style = MaterialTheme.typography.body2
                )
            }
        }

        Text(episode.air_date, modifier = Modifier.padding(start = 16.dp))
    }
    Divider()
}

File: iosApp/iosApp/Features/Episodes/EpisodesListRowView.swift
import SwiftUI
import shared


struct EpisodesListRowView: View {
    let episode: EpisodeDetail
    
    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack {
                VStack(alignment: .leading, spacing: 4) {
                    Text(episode.name)
                        .font(.title3)
                        .fontWeight(.bold)
                        .foregroundColor(.primary)
                        .lineLimit(1)
                    
                    Text(episode.episode)
                        .font(.subheadline)
                        .foregroundColor(.secondary)
                }
                
                Spacer()
                
                Text(episode.air_date)
                    .font(.caption)
                    .foregroundColor(.secondary)
                    .padding(.leading, 8)
            }
            
            Divider()
                .padding(.top, 4)
        }
        .padding(.vertical, 4)
    }
}
